tests/migrations/test_autodetector.py

```diff
@@ -652,18 +652,16 @@ def test_alter_field(self):
         self.assertOperationTypes(changes, 'testapp', 0, ["AlterField"])
         self.assertOperationAttributes(changes, "testapp", 0, 0, name="name", preserve_default=True)
 
-    def test_alter_field_to_not_null_with_default(self):
+    @mock.patch('django.db.migrations.questioner.MigrationQuestioner.ask_not_null_alteration',
+                side_effect=Exception("Should not have prompted for not null addition"))
```

Comment:
Small nitpick, I would replace those bare side effect `Exception` by `AssertionError` while you're around.